lib/legacy/class.html

```diff
@@ -133,31 +143,103 @@
       return list;
     }
 
+    function mergePropertyInfo(a, b) {
+      if (typeof b === 'function') {
+        a.type = b;
+        return;
+      }
+      if ('type' in b) {
+        a.type = b.type;
+      }
+      if ('value' in b) {
+        a.value = b.value;
+      }
+      // readOnly: cannot become false and takes on `computed` value
+      a.readOnly = a.readOnly || Boolean(a.computed) || b.readOnly || Boolean(b.computed);
+      // computed: first in wins
+      if ('computed' in b) {
+        a.computed = a.computed || b.computed;
+      }
+      // reflectToAttribute: cannot become false
+      if ('reflectToAttribute' in b) {
+        a.reflectToAttribute = a.reflectToAttribute || b.reflectToAttribute;
+      }
+      // notify: cannot become false (only install 1x)
+      if ('notify' in b) {
+        a.notify = a.notify || b.notify;
+      }
+      // make this an array
+      if ('observer' in b) {
+        if (!a.observer) {
+          a.observer = b.observer;
+        // we have both: make array and push onto
+        } else {
+          if (!Array.isArray(a.observer)) {
+            a.observer = [a.observer];
+          }
+          a.observer.push(b.observer);
+        }
+      }
+    }
+
+    function mergeProperties(a, b) {
+      for (let p in b) {
+        a[p] = a[p] || {};
+        mergePropertyInfo(a[p], b[p]);
+      }
+    }
+
     /**
      * @param {!PolymerInit} info Polymer info object
      * @param {function(new:HTMLElement)} Base base class to extend with info object
+     * @param {Object} behaviors behaviors to copy into the element
      * @return {function(new:HTMLElement)} Generated class
      * @suppress {checkTypes}
      * @private
      */
-    function GenerateClassFromInfo(info, Base) {
+    function GenerateClassFromInfo(info, Base, behaviors) {
 
       /** @private */
       class PolymerGenerated extends Base {
 
         static get properties() {
-          return info.properties;
+          const properties = {};
+          if (this.prototype.__behaviors) {
+            for (let i=0, b; i < this.prototype.__behaviors.length; i++) {
+              b = this.prototype.__behaviors[i];
+              mergeProperties(properties, b.properties);
+            }
+          }
+          mergeProperties(properties, info.properties);
```

Comment:
It seemed like a possible optimization to skip the `mergeProperties` step from the base info object into an empty object; it seems superfulous for the initial class -- but then I realized this was required for the `computed` to `readOnly` coersion and the shorthand type expansion.  So that was not intuitive, maybe due to the naming?  Suggest different name or at least a comment.

Reply:
Added comment.